src/Tests/Sending/MessageDispatcherTests.cs

```diff
@@ -470,13 +448,41 @@ public async Task Should_split_into_batches_of_max_4500_when_no_transactions_use
 
             await dispatcher.Dispatch(new TransportOperations(operations.ToArray()), azureServiceBusTransaction.TransportTransaction);
 
-            Assert.That(defaultSender.BatchSentMessages, Has.Count.EqualTo(3));
+            Assert.That(defaultSender.BatchSentMessages, Has.Count.EqualTo(2));
             var firstBatch = defaultSender[defaultSender.BatchSentMessages.ElementAt(0)];
             var secondBatch = defaultSender[defaultSender.BatchSentMessages.ElementAt(1)];
-            var thirdBatch = defaultSender[defaultSender.BatchSentMessages.ElementAt(2)];
-            Assert.That(firstBatch, Has.Count.EqualTo(4500));
+            Assert.That(firstBatch, Has.Count.EqualTo(150));
             Assert.That(secondBatch, Has.Count.EqualTo(50));
-            Assert.That(thirdBatch, Has.Count.EqualTo(50));
+        }
+
+        [Test]
+        public async Task Should_fallback_to_individual_sends_when_messages_cannot_be_added_to_batch()
```

Comment:
This is a first blunt test. Of course it might be good to add a bit more complex test where during batching one exceeds the batch limitations.